### PR TITLE
SUM-113 - Card variant

### PR DESCRIPTION
**Pull request by mdyoung3**
# READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- Add card variant: https://stanfordits.atlassian.net/browse/SUM-113

# Review By (Date)
- When does this need to be reviewed by?

# Urgency
- How critical is this PR?

# Steps to Test

1. Do this
1. Then this
2. Then this

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   An updated associative array representing the form structure with the
+   *   added 'card_variant' element.
+   */
+  public function buildBehaviorForm(ParagraphInterface $paragraph, array &$form, FormStateInterface $form_state): array {
+    $element = parent::buildBehaviorForm($paragraph, $form, $form_state);
+    $element['card_variant'] = [
```
Comment (pookmish) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: Let's name this something `sum_`

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      '#type' => 'select',
+      '#title' => $this->t('Card variant'),
```
Comment (pookmish) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php:34]: Add `'#default_option' = $this->t('Default'),`

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      '#options' => [
```
Comment (pookmish) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php:36]: needs `#default_value`

**Review by pookmish — CHANGES_REQUESTED** [review]
A couple quick fixes needed

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php (hunk 3 above)
Comment (pookmish) [reply docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php:36]: `#default_value` Still needed

**Review by pookmish — CHANGES_REQUESTED** [review]

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php (hunk 2 above)
Comment (mdyoung3) [reply docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php:34]: @pookmish Did you mean #empty_option?

Comment (pookmish) [reply docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php:34]: Sorry, yeah it's `empty_option`

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      '#type' => 'select',
+      '#title' => $this->t('Card variant'),
+      '#default_option' => $this->t('Default'),
+      '#options' => [
+        'circle' => $this->t('Circle'),
+        'semicircle-top' => $this->t('Semicircle top'),
+        'semicircle-bottom' => $this->t('Semicircle bottom'),
+        'oval' => $this->t('Oval'),
+      ],
+      '#default_value' => 'oval',
```
Comment (pookmish) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: You need to get the value from the paragraph, you can't car code the default value. Besides, the default should be `null` Review the parent class to see how it's done there. you can copy from there.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+      '#type' => 'select',
+      '#title' => $this->t('Card variant'),
+      '#empty_option' => $this->t('Default'),
+      '#options' => [
+        'circle' => $this->t('Circle'),
+        'semicircle-top' => $this->t('Semicircle top'),
+        'semicircle-bottom' => $this->t('Semicircle bottom'),
+        'oval' => $this->t('Oval'),
+      ],
+      '#default_value' => $paragraph->getBehaviorSetting('sum_card_style', 'sum_card_variant'),
```
Comment (mdyoung3) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: @pookmish Is the plugin_id defined here correctly?

Comment (pookmish) [reply docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: no, you use `su_card_styles`

File: docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    $element = parent::buildBehaviorForm($paragraph, $form, $form_state);
+    $element['sum_card_variant'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Card variant'),
+      '#empty_option' => $this->t('Default'),
+      '#options' => [
+        'circle' => $this->t('Circle'),
+        'semicircle-top' => $this->t('Semicircle top'),
+        'semicircle-bottom' => $this->t('Semicircle bottom'),
+        'oval' => $this->t('Oval'),
```
Comment (pookmish) [inline docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: Ok, we got syntax issues out of the way. Now these options, i'm not sure. I only know of the regular rectangle (Default), and the oval. Where do you see the other three options in the designs?

Comment (mdyoung3) [reply docroot/profiles/lagunita/summer_profile/modules/summer_helper/src/Plugin/paragraphs/Behavior/SummerCardBehaviors.php]: @pookmish Oops. Wrong figam file. I checked with Kat. You're correct.

**Review by pookmish — APPROVED** [review]
Thanks

**Pull request merged**